collect_app/src/test/java/org/odk/collect/android/formentry/repeats/DeleteRepeatDialogFragmentTest.java

```diff
@@ -0,0 +1,131 @@
+    }
+
+    @Test
+    public void fragmentActivityShouldImplementDeleteRepeatDialogCallback() {
+        dialogFragment.show(fragmentManager, "TAG");
+        assertThat(dialogFragment.getActivity(), instanceOf(DeleteRepeatDialogFragment.DeleteRepeatDialogCallback.class));
+    }
+
+    @Test
+    public void dialogIsNotCancellable() {
+        dialogFragment.show(fragmentManager, "TAG");
+        assertThat(shadowOf(dialogFragment.getDialog()).isCancelable(), equalTo(false));
+    }
+
+    @Test
+    public void shouldShowCorrectMessage() {
+        when(formEntryViewModel.getLastRepeatedGroupName()).thenReturn("blah");
+        when(formEntryViewModel.getLastRepeatedGroupRepeatCount()).thenReturn(0);
+        dialogFragment.show(fragmentManager, "TAG");
+        AlertDialog dialog = (AlertDialog) ShadowDialog.getLatestDialog();
+        String message = ((TextView) dialog.findViewById(android.R.id.message)).getText().toString();
+
+        assertThat(message, equalTo(RuntimeEnvironment.application.getString(R.string.delete_repeat_confirm, "blah (1)")));
+    }
+
+    @Test
+    public void clickingCancel_shouldDismissTheDialog() {
+        dialogFragment.show(fragmentManager, "TAG");
+        AlertDialog dialog = (AlertDialog) ShadowDialog.getLatestDialog();
+        assertTrue(dialog.isShowing());
+
+        dialog.getButton(DialogInterface.BUTTON_NEGATIVE).performClick();
+        assertFalse(dialog.isShowing());
+        assertTrue(shadowOf(dialog).hasBeenDismissed());
+    }
+
+    @Test
+    public void clickingRemoveGroup_shouldDismissTheDialog() {
+        dialogFragment.show(fragmentManager, "TAG");
+        AlertDialog dialog = (AlertDialog) ShadowDialog.getLatestDialog();
+        assertTrue(dialog.isShowing());
+
+        dialog.getButton(DialogInterface.BUTTON_POSITIVE).performClick();
+        assertFalse(dialog.isShowing());
+        assertTrue(shadowOf(dialog).hasBeenDismissed());
+    }
+
+    @Test
+    public void clickingRemoveGroup_callsDeleteGroup() {
+        dialogFragment.show(fragmentManager, "TAG");
+        AlertDialog dialog = (AlertDialog) ShadowDialog.getLatestDialog();
+        assertThat(activity.deleteGroupCalled, equalTo(false));
+
+        dialog.getButton(DialogInterface.BUTTON_POSITIVE).performClick();
+        assertThat(activity.deleteGroupCalled, equalTo(true));
+    }
+
+    @Test
+    public void clickingCancel_callsOnCancelled() {
+        dialogFragment.show(fragmentManager, "TAG");
+        AlertDialog dialog = (AlertDialog) ShadowDialog.getLatestDialog();
+        assertThat(activity.onCancelledCalled, equalTo(false));
+
+        dialog.getButton(DialogInterface.BUTTON_NEGATIVE).performClick();
+        assertThat(activity.onCancelledCalled, equalTo(true));
+    }
+
+    public static class TestActivity extends FragmentActivity implements DeleteRepeatDialogFragment.DeleteRepeatDialogCallback {
+
```

Comment:
@lognaturel I changed the tests a little bit so as to avoid using protected scope for callback. I like this approach better than directly approaching the interface in the fragment.